### ISSUE TITLE
Loading subroute in conjunction with equilevel index route produce incorrect entries in the browser history

### ISSUE THREAD

**Issue by azhiv** [ember-cli/ember-page-title#282]
Checkout this routes structure:
<img width="209" alt="Screenshot 2024-01-22 at 17 32 28" src="https://github.com/ember-cli/ember-page-title/assets/32125472/6c937edd-d655-4e05-80b5-0fff008d993d">

When I navigate from the root app route down to Project down to Store, then back up to Project and finally find myself on the app route again, I would expect this full path reflected in the browser history. However, this is what I see there:

<img width="319" alt="Screenshot 2024-01-22 at 17 35 03" src="https://github.com/ember-cli/ember-page-title/assets/32125472/eb17ed4e-ceb4-4409-ba1f-d19f650049c6">

Here's how the picture starts looking different (and correct) when I delete the `loading.hbs` subroute highlighted green in the first screenshot:
<img width="315" alt="Screenshot 2024-01-22 at 17 37 39" src="https://github.com/ember-cli/ember-page-title/assets/32125472/ca57f5fb-a01e-4884-8ad5-1e020271197d">

I can also confirm that deleting the `index` subroute (without touching the `loading` one) also produces the correct history:
<img width="314" alt="Screenshot 2024-01-22 at 17 43 39" src="https://github.com/ember-cli/ember-page-title/assets/32125472/a9a10d6f-cf13-4d5e-8d31-94b53f470d9a">

So to conclude, having both `index` and `loading` subroutes make the framework generate incorrect entries in the browser history while removing either of them makes things look correct. [Here's](https://github.com/azhiv/ember-page-title-repro) the reproduction repo.

**Comment by sergekoval** [comment]
noticed the same behavior as well. We've been trying to replace our custom page title implementation with this addon to have more clean solution but unfortunately it creates incorrect history in browser.

Original implementation:

![image](https://github.com/ember-cli/ember-page-title/assets/6385650/cc49dd88-f1a0-4ebe-9592-5cfe77bd2431)

Using this addon:

![image](https://github.com/ember-cli/ember-page-title/assets/6385650/6e583530-a801-455b-adfb-6b13692bd36e)

Basically as mentioned before, loading transitions reset the tab title with default app name which registers in the history instead of actual following sub route.
